pkg/generators/models/models.go

```diff
@@ -238,14 +238,40 @@ func resolveAllOf(ref *openapi3.SchemaRef, passed passedSchemas) (out *openapi3.
 		passed = make(passedSchemas)
 	}
 
+	// very special case where the allOf references
+	// another named schema but is not actually merging
+	// several schemas, this is usually just overriding
+	// a value like nullable or the description
```

Comment:
to add just a little more context here, this is here to enable a special case we often want to use in our own apis

We have been writing some types like this
```yaml
CreateRequest:
    type: object
    properties:
        name: string
        status:
          allOf:
            - $ref: "#/components/schemas/LifecycleStatus"
            - nullable: true
```

Expecting the output to be 

```go
type CreateRequest struct {
    Name string
    Status *LifecycleStatus
}
```

However, the new/correct implementation for deep merging the values in the allOf would produce this instead
```go
type CreateRequest struct {
    Name string
    Status *string
}
```
the named enum is gone now be the deep merge strips out the references and leaves only the underyling type definition.

However, this yaml schema is an equivalent and valid alternative to the one above

```yaml
CreateRequest:
    type: object
    properties:
        name: string
        status:
          nullable: true
          allOf:
            - $ref: "#/components/schemas/LifecycleStatus"
  ```

So i added a special case that looks for `allOf` with exactly one item. it then skips the deep merge step, which leaves the original reference and produces the desired Go type

```go
type CreateRequest struct {
    Name string
    Status *LifecycleStatus
}
```


It is important to note that the original schema is still valid and will still be parsed by the generator and produce a valid Go type. The main difference between the two results is that the first produces a field that has no validation (because it doesn't reference the actual enum) while the second produces the correct validation methods, because it still references the enum.  

So the result is that OpenAPI treats the two schemas as equivalent, but our generator can only create the correct validation code for the the second schema.